### ISSUE TITLE
Last import directory not stored

### ISSUE THREAD

**Issue by jymen** [UnderwaterApps/overlap2d#97]
Usually you put your stuff(images,fonts,particles...) in a dedicated game resource directory
=> It would be nice to keep the last used import directory location for next import session

**Comment by azakhary** [comment]
I even go further and say we need a new importing mechanism.

several things to consider:
1) would be awesome if we support drag dropping from outside
2) some kind of pre-import mode, so that you drag drop all you need, and when you are ready you click - import.

**Comment by azakhary** [comment]
New Importing mechanism starting from #104 
Although this issue still stands, but easy to fix. I will take on it.

**Comment by jymen** [comment]
I make a first test : definitively better promising .... Although
- Drag Drop from finder does not work on macosx.(Not sure but this feature may request some external native mecanism ...)
- Importing simple image from tree panel is Ok 
- I wonder how to import Texture (selecting both .png+.txt does not seem to work)

**Comment by azakhary** [comment]
When you say import Texture - you mean import atlas pack with many texture regions in it? If so. this one is marked as "todo" and I'll add it shortly.

For osx - that sucks, I do not have all set up on my macbook, I was kinda hoping it would work. I think it should, probably some kind of small issue, has to be debugged on mac. If anyone with sources and osx can check this out would be awesome.

**Comment by jymen** [comment]
Correct I mean atlas pack with many textures.
Ok I'll have a look at  Drop files in java apps coming from Mac finder  ... This may be a small issue ... or Not  ( I am currious to know if this feature is also KO on linux ? )

**Comment by azakhary** [comment]
in Main class I just put it inside the if that excludes osx imho, the problem was that for osx we use LwjglApplication instead of LwjglFrame so that there was no frame I could attach this event to.  If you can find a way to get the frame from LwjglApp class and also add that line to the osx if statement. things should work. I am talking about that line in Main class that sets DragDropListener

**Comment by jymen** [comment]
Dragging files from finder and friends on osx requests 2 extra things : 
1) using the com.apple.eawt.OpenFilesHandler class which will be triggered when a finder file is dropped on a given app.
2) Package the jar in a full standard macos app and granting external file access in the info.plist  file

Some work to be done : I can handle that if you're ok => let me know

**Comment by azakhary** [comment]
Yeah that would be awesomolous if you can take on that one!

**Comment by azakhary** [comment]
Hey but can you recheck again, as far as I now Java dnd library (java.awt.dnd.) should work on osx, the only reason it doesnot in o2d is that it is deliberately not in the if statement of osx, and that to be in the other if statement we have to get the frame from LwjglApplication I think that should be enough for jar to work and dnd. Now need to do extra work here.

**Comment by azakhary** [comment]
Or if you have a link to article that says that java.awt,dnd doesnot work on osx, please share

**Comment by jymen** [comment]
I already made a test yesterday by putting the DropTarget dropTarget = new DropTarget(mainFrame, new FileDropListener()); in the osX part but drop event is not triggered then. So my guess is that the drop event will not be triggered until your java code is not bundled into a mac app bundle : 
although following article is a bit old it's still accurate : 
https://developer.apple.com/library/mac/documentation/Java/Conceptual/Java14Development/03-JavaDeployment/JavaDeployment.html
This one about double click on start implies also to be in a bundle to have files events being populated by Cocoa to an app
http://ruby.boodlefish.com/2013/04/apple-java-extensions-openfiles.html.
Not sure about wether java.awt.dnd or com.apple.eawt.OpenFilesHandler is triggered then but I think that to be triggered your app needs to be part of the Cocoa ecosystem by being packaged in a Macos bundle
I continue to make some test about this and let you know my final conclusion later

**Comment by azakhary** [comment]
I  see, might be the case. I can see already dived deep in this. So I guess you know better :D I will also try to read some stuff today, see if I can be of any help.

**Comment by azakhary** [comment]
(But if it true and we can't use jar but have to provide other executable, then gradle distribution script has to modify as well to do that automatically) and I will have to modify website to give correct executable.

**Comment by jymen** [comment]
I agree about gradle changes then ; perhaps using the libgdx portable tool for that may help: 
https://github.com/libgdx/packr
Providing executable is also better for end users rather than an executable jar (Better os integration) even if it's not urgent stuff

**Comment by azakhary** [comment]
Makes sense. This had to be done at one point anyway. I just didn't want to blow a simple feature request into something too big :))

**Comment by jymen** [comment]
I agree as a first step I will just check that 'bundling' inside an osx apps solves the drag drop file for osx and let you know

**Comment by azakhary** [comment]
Awesome, thanks a lot!

**Comment by jymen** [comment]
I made some progress : 
- I have built a simplified sample with a DropTargetListener and it works with file drops from finder both with bundle app and without bundle app => so the problem is not linked with osx bundling ... Good news.
- will now continue comparing my sample code with Overlap2d code to see where the problem stands ... I think it should not be too complicated ...

**Comment by azakhary** [comment]
how do you get the frame for the listener? (or what do you pass to the listener?)

**Comment by jymen** [comment]
public class DropTest extends JFrame implements DropTargetListener {

  DropTarget dt;
  JTextArea ta = new JTextArea();

  public DropTest() {
    super("Drop Test");

    // Handle mac-specific events (if we're running under OS X).
    Application macApplication = Application.getApplication();
    OSXHandler osxHandler = new OSXHandler();
    macApplication.setOpenFileHandler(osxHandler);

    System.err.println("Registered application");
    add(new JLabel("Drop a list from your file chooser here:"), BorderLayout.NORTH);
    ta.setBackground(Color.white);
    getContentPane().add(ta, BorderLayout.CENTER);

    dt = new DropTarget(ta, this);
    setTitle("Main");
    setSize(300, 200);
    setLocationRelativeTo(null);
    setDefaultCloseOperation(EXIT_ON_CLOSE);
  }

**Comment by azakhary** [comment]
No, I meant in the o2d example, not the test.

**Comment by jymen** [comment]
I just copied the dropTarget init from the else part inside the osx part : 
            // subscribe to file dropping notifications, currently windows only
            DropTarget dropTarget = new DropTarget(mainFrame, new FileDropListener());

**Comment by azakhary** [comment]
That's why it doesnot work :) as I mentioned earlier,  there is no mainFrame in the osx "if" as different way is used. that's the main issue there.

**Comment by azakhary** [comment]
in windows exaple there is
mainFrame = new LwjglFrame(overlap2D, config);

But in osx
new JglfwApplication(overlap2D, config);

the trick is to find way to retrieve some kinf of Frame instance from JglfwApplication in order to attach listener to it.

**Comment by jymen** [comment]
OK I got it I think that the solution for osx may consist instead of using a  LwjglCanvas instead of  JglfwApplication and to wrap it into Swing Jframe content pane launched at Osx test startup something like : 
 LwjglCanvas canvas = new LwjglCanvas(myapp, myconfig);
 SwingUtilities.invokeLater(
new Runnable()
    {
       public void run()
       {
           myFrame.getContentPane().add(canvas.getCanvas());
       }
    }
 );

I can test this kind of solution on osx sometimes this WE if you agree

**Comment by azakhary** [comment]
The reason we were using JglfwApplication, is because the other one goes crazy on osx CPU
